rtgui/previewhandler.cc

```diff
@@ -204,27 +204,19 @@ Glib::RefPtr<Gdk::Pixbuf> PreviewHandler::getRoughImage (int x, int y, int w, in
             h = image->getHeight() * totalZoom;
         }
 
-        int ix = x * zoom;
-        int iy = y * zoom;
+        x *= zoom;
+        y *= zoom;
 
-        if (ix < 0) {
-            ix = 0;
+        if ((x + w) / totalZoom > previewImg->get_width()) {
```

Comment:
I don't mind the casts, but the inner one should not be necessary as `int * double` should automatically result in `double`.